--- server/src/utils/common.ts
+++ server/src/server.ts
@@ -1,22 +1,22 @@
-function doAddDataExportTask(

-  math_env: string | undefined,

-  email: string,

-  zid: number,

-  atDate: number,

-  format: string,

-  task_bucket: number

-) {

-  return pgQueryP(

-    "insert into worker_tasks (math_env, task_data, task_type, task_bucket) values ($1, $2, 'generate_export_data', $3);",

-    [

-      math_env,

-      {

-        email: email,

-        zid: zid,

-        "at-date": atDate,

-        format: format,

-      },

-      task_bucket, // TODO hash the params to get a consistent number?

-    ]

-  );

-}

+  function doAddDataExportTask(

+    math_env: string | undefined,

+    email: string,

+    zid: number,

+    atDate: number,

+    format: string,

+    task_bucket: number

+  ) {

+    return pgQueryP(

+      "insert into worker_tasks (math_env, task_data, task_type, task_bucket) values ($1, $2, 'generate_export_data', $3);",

+      [

+        math_env,

+        {

+          email: email,

+          zid: zid,

+          "at-date": atDate,

+          format: format,

+        },

+        task_bucket, // TODO hash the params to get a consistent number?

+      ]

+    );

+  }

