--- server/src/routes/password.ts
+++ server/src/server.ts
@@ -1,59 +1,59 @@
-function handle_POST_auth_password(

-  req: { p: { pwresettoken: any; newPassword: any } },

-  res: {

-    status: (

-      arg0: number

-    ) => {

-      (): any;

-      new (): any;

-      json: { (arg0: string): void; new (): any };

-    };

+  function handle_POST_auth_password(

+    req: { p: { pwresettoken: any; newPassword: any } },

+    res: {

+      status: (

+        arg0: number

+      ) => {

+        (): any;

+        new (): any;

+        json: { (arg0: string): void; new (): any };

+      };

+    }

+  ) {

+    let pwresettoken = req.p.pwresettoken;

+    let newPassword = req.p.newPassword;

+

+    getUidForPwResetToken(

+      pwresettoken,

+      //     Argument of type '(err: any, userParams: { uid?: any; }) => void' is not assignable to parameter of type '(arg0: number | null, arg1?: { uid: any; } | undefined) => void'.

+      // Types of parameters 'userParams' and 'arg1' are incompatible.

+      //   Type '{ uid: any; } | undefined' is not assignable to type '{ uid?: any; }'.

+      //     Type 'undefined' is not assignable to type '{ uid?: any; }'.ts(2345)

+      // @ts-ignore

+      function (err: any, userParams: { uid?: any }) {

+        if (err) {

+          fail(

+            res,

+            500,

+            "Password Reset failed. Couldn't find matching pwresettoken.",

+            err

+          );

+          return;

+        }

+        let uid = Number(userParams.uid);

+        generateHashedPassword(

+          newPassword,

+          function (err: any, hashedPassword: any) {

+            return pgQueryP(

+              "insert into jianiuevyew (uid, pwhash) values " +

+                "($1, $2) on conflict (uid) " +

+                "do update set pwhash = excluded.pwhash;",

+              [uid, hashedPassword]

+            ).then(

+              (rows: any) => {

+                res.status(200).json("Password reset successful.");

+                clearPwResetToken(pwresettoken, function (err: any) {

+                  if (err) {

+                    logger.error("polis_err_auth_pwresettoken_clear_fail", err);

+                  }

+                });

+              },

+              (err: any) => {

+                fail(res, 500, "Couldn't reset password.", err);

+              }

+            );

+          }

+        );

+      }

+    );

   }

-) {

-  let pwresettoken = req.p.pwresettoken;

-  let newPassword = req.p.newPassword;

-

-  getUidForPwResetToken(

-    pwresettoken,

-    //     Argument of type '(err: any, userParams: { uid?: any; }) => void' is not assignable to parameter of type '(arg0: number | null, arg1?: { uid: any; } | undefined) => void'.

-    // Types of parameters 'userParams' and 'arg1' are incompatible.

-    //   Type '{ uid: any; } | undefined' is not assignable to type '{ uid?: any; }'.

-    //     Type 'undefined' is not assignable to type '{ uid?: any; }'.ts(2345)

-    // @ts-ignore

-    function (err: any, userParams: { uid?: any }) {

-      if (err) {

-        fail(

-          res,

-          500,

-          "Password Reset failed. Couldn't find matching pwresettoken.",

-          err

-        );

-        return;

-      }

-      let uid = Number(userParams.uid);

-      generateHashedPassword(

-        newPassword,

-        function (err: any, hashedPassword: any) {

-          return pgQueryP(

-            "insert into jianiuevyew (uid, pwhash) values " +

-              "($1, $2) on conflict (uid) " +

-              "do update set pwhash = excluded.pwhash;",

-            [uid, hashedPassword]

-          ).then(

-            (rows: any) => {

-              res.status(200).json("Password reset successful.");

-              clearPwResetToken(pwresettoken, function (err: any) {

-                if (err) {

-                  logger.error("polis_err_auth_pwresettoken_clear_fail", err);

-                }

-              });

-            },

-            (err: any) => {

-              fail(res, 500, "Couldn't reset password.", err);

-            }

-          );

-        }

-      );

-    }

-  );

-}

