--- server/src/routes/math.ts
+++ server/src/server.ts
@@ -1,95 +1,95 @@
-function handle_GET_math_correlationMatrix(

-  req: { p: { rid: any; math_tick: any } },

-  res: {

-    status: (

-      arg0: number

-    ) => {

-      (): any;

-      new (): any;

-      json: { (arg0: { status: string }): void; new (): any };

-    };

-    json: (arg0: any) => void;

+  function handle_GET_math_correlationMatrix(

+    req: { p: { rid: any; math_tick: any } },

+    res: {

+      status: (

+        arg0: number

+      ) => {

+        (): any;

+        new (): any;

+        json: { (arg0: { status: string }): void; new (): any };

+      };

+      json: (arg0: any) => void;

+    }

+  ) {

+    let rid = req.p.rid;

+    let math_env = Config.mathEnv;

+    let math_tick = req.p.math_tick;

+

+    function finishAsPending() {

+      res.status(202).json({

+        status: "pending",

+      });

+    }

+

+    function hasCommentSelections() {

+      return pgQueryP(

+        "select * from report_comment_selections where rid = ($1) and selection = 1;",

+        [rid]

+        // Argument of type '(rows: string | any[]) => boolean' is not assignable to parameter of type '(value: unknown) => boolean | PromiseLike<boolean>'.

+        // Types of parameters 'rows' and 'value' are incompatible.

+        // Type 'unknown' is not assignable to type 'string | any[]'.

+        //     Type 'unknown' is not assignable to type 'any[]'.ts(2345)

+        // @ts-ignore

+      ).then((rows: string | any[]) => {

+        return rows.length > 0;

+      });

+    }

+

+    let requestExistsPromise = pgQueryP(

+      "select * from worker_tasks where task_type = 'generate_report_data' and math_env=($2) " +

+        "and task_bucket = ($1) " +

+        // "and attempts < 3 " +

+        "and (task_data->>'math_tick')::int >= ($3) " +

+        "and finished_time is NULL;",

+      [rid, math_env, math_tick]

+    );

+

+    let resultExistsPromise = pgQueryP(

+      "select * from math_report_correlationmatrix where rid = ($1) and math_env = ($2) and math_tick >= ($3);",

+      [rid, math_env, math_tick]

+    );

+

+    Promise.all([resultExistsPromise, getZidForRid(rid)])

+      .then((a: any[]) => {

+        let rows = a[0];

+        let zid = a[1];

+        if (!rows || !rows.length) {

+          //         Argument of type '(requests_rows: string | any[]) => globalThis.Promise<void> | undefined' is not assignable to parameter of type '(value: unknown) => void | PromiseLike<void | undefined> | undefined'.

+          // Types of parameters 'requests_rows' and 'value' are incompatible.

+          //   Type 'unknown' is not assignable to type 'string | any[]'.

+          //           Type 'unknown' is not assignable to type 'any[]'.ts(2345)

+          // @ts-ignore

+          return requestExistsPromise.then((requests_rows: string | any[]) => {

+            const shouldAddTask = !requests_rows || !requests_rows.length;

+            // const shouldAddTask = true;

+

+            if (shouldAddTask) {

+              return hasCommentSelections().then((hasSelections: any) => {

+                if (!hasSelections) {

+                  return res.status(202).json({

+                    status: "polis_report_needs_comment_selection",

+                  });

+                }

+                return pgQueryP(

+                  "insert into worker_tasks (task_type, task_data, task_bucket, math_env) values ('generate_report_data', $1, $2, $3);",

+                  [

+                    JSON.stringify({

+                      rid: rid,

+                      zid: zid,

+                      math_tick: math_tick,

+                    }),

+                    rid,

+                    math_env,

+                  ]

+                ).then(finishAsPending);

+              });

+            }

+            finishAsPending();

+          });

+        }

+        res.json(rows[0].data);

+      })

+      .catch((err: any) => {

+        return fail(res, 500, "polis_err_GET_math_correlationMatrix", err);

+      });

   }

-) {

-  let rid = req.p.rid;

-  let math_env = Config.mathEnv;

-  let math_tick = req.p.math_tick;

-

-  function finishAsPending() {

-    res.status(202).json({

-      status: "pending",

-    });

-  }

-

-  function hasCommentSelections() {

-    return pgQueryP(

-      "select * from report_comment_selections where rid = ($1) and selection = 1;",

-      [rid]

-      // Argument of type '(rows: string | any[]) => boolean' is not assignable to parameter of type '(value: unknown) => boolean | PromiseLike<boolean>'.

-      // Types of parameters 'rows' and 'value' are incompatible.

-      // Type 'unknown' is not assignable to type 'string | any[]'.

-      //     Type 'unknown' is not assignable to type 'any[]'.ts(2345)

-      // @ts-ignore

-    ).then((rows: string | any[]) => {

-      return rows.length > 0;

-    });

-  }

-

-  let requestExistsPromise = pgQueryP(

-    "select * from worker_tasks where task_type = 'generate_report_data' and math_env=($2) " +

-      "and task_bucket = ($1) " +

-      // "and attempts < 3 " +

-      "and (task_data->>'math_tick')::int >= ($3) " +

-      "and finished_time is NULL;",

-    [rid, math_env, math_tick]

-  );

-

-  let resultExistsPromise = pgQueryP(

-    "select * from math_report_correlationmatrix where rid = ($1) and math_env = ($2) and math_tick >= ($3);",

-    [rid, math_env, math_tick]

-  );

-

-  Promise.all([resultExistsPromise, getZidForRid(rid)])

-    .then((a: any[]) => {

-      let rows = a[0];

-      let zid = a[1];

-      if (!rows || !rows.length) {

-        //         Argument of type '(requests_rows: string | any[]) => globalThis.Promise<void> | undefined' is not assignable to parameter of type '(value: unknown) => void | PromiseLike<void | undefined> | undefined'.

-        // Types of parameters 'requests_rows' and 'value' are incompatible.

-        //   Type 'unknown' is not assignable to type 'string | any[]'.

-        //           Type 'unknown' is not assignable to type 'any[]'.ts(2345)

-        // @ts-ignore

-        return requestExistsPromise.then((requests_rows: string | any[]) => {

-          const shouldAddTask = !requests_rows || !requests_rows.length;

-          // const shouldAddTask = true;

-

-          if (shouldAddTask) {

-            return hasCommentSelections().then((hasSelections: any) => {

-              if (!hasSelections) {

-                return res.status(202).json({

-                  status: "polis_report_needs_comment_selection",

-                });

-              }

-              return pgQueryP(

-                "insert into worker_tasks (task_type, task_data, task_bucket, math_env) values ('generate_report_data', $1, $2, $3);",

-                [

-                  JSON.stringify({

-                    rid: rid,

-                    zid: zid,

-                    math_tick: math_tick,

-                  }),

-                  rid,

-                  math_env,

-                ]

-              ).then(finishAsPending);

-            });

-          }

-          finishAsPending();

-        });

-      }

-      res.json(rows[0].data);

-    })

-    .catch((err: any) => {

-      return fail(res, 500, "polis_err_GET_math_correlationMatrix", err);

-    });

-}

